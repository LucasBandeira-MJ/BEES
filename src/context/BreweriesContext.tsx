import { createContext, useState } from "react";


interface BreweriesContextData {
    user: string
    updateUser: (username: string) => void
}

interface BreweriesProviderProps {
    children: React.ReactNode
}

export const BreweriesContext = createContext({} as BreweriesContextData)

export const BreweriesProvider = ({children}: BreweriesProviderProps) => {
    const [user, setUser] = useState<string>('')
    
    const updateUser = (username: string) => {
        setUser(username)
    }

    return (
        <BreweriesContext.Provider value={{
            user,
            updateUser
        }}>
            { children }
        </BreweriesContext.Provider>
    )
}