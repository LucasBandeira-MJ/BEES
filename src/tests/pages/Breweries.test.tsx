import { render } from "@testing-library/react"
import { mocked } from 'jest-mock'
import { useRouter } from 'next/router'

import renderer from 'react-test-renderer'

import { BreweriesContext } from "../../context/BreweriesContext"
import Breweries from '../../pages/breweries'

jest.mock('next/router')

describe("Breweries page", () => {

    it('should match the snapshot', () => {
        const snapshot = renderer
            .create(<Breweries />)
            .toJSON()

            expect(snapshot).toMatchSnapshot()
    })

    it("should redirect user to Home page if user is not set", () => {
        
        const useRouterMocked = mocked(useRouter) 
        const pushMocked = jest.fn()

        useRouterMocked.mockReturnValueOnce({
            push: pushMocked
        } as any)

        render(
            <BreweriesContext.Provider value={{ user: ''} as any}>
                <Breweries />
            </BreweriesContext.Provider>
        )

        expect(pushMocked).toHaveBeenCalled()  
    })
})