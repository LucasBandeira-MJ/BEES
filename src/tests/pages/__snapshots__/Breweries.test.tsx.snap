// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breweries page should match the snapshot 1`] = `
<div
  className="sc-gsnTZi favrQF"
>
  <header
    className="sc-bczRLJ haxtDN"
  >
    <div
      data-testid="go-back"
    >
      <img
        alt="back arrow"
        src="images/arrow-circle-left.svg"
      />
      <strong>
        Go Back
      </strong>
    </div>
    <strong />
  </header>
  <main />
</div>
`;
